cmd/hemictl/hemictl.go

```diff
@@ -1142,12 +1232,182 @@ func HandleSignals(ctx context.Context, cancel context.CancelFunc, callback func
 	os.Exit(2)
 }
 
-func _main() error {
-	if len(os.Args) < 2 {
-		usage()
-		return errors.New("not enough parameters")
+func IsJSON(str string) bool {
+	var js json.RawMessage
+	return json.Unmarshal([]byte(str), &js) == nil
+}
+
+func Jsonify(args []string) (string, error) {
+	formatted := "{"
+	for i, c := range args {
+		if i != 0 {
+			formatted += ","
+		}
+		kv := strings.SplitN(c, "=", 2)
+		if len(kv) != 2 {
+			return formatted, fmt.Errorf("invalid argument format: %v", c)
+		}
+		formatted = fmt.Sprintf("%s\"%s\": %v", formatted, kv[0], kv[1])
 	}
+	formatted += "}"
+
+	return formatted, nil
+}
+
+func parsePayload(cmd string, args []string) (any, error) {
+	cmdType, ok := allCommands[cmd]
+	if !ok {
+		return nil, fmt.Errorf("unknown command: %v", cmd)
+	}
+
+	clone := reflect.New(cmdType).Interface()
+	log.Debugf("%v", spew.Sdump(clone))
+	if len(args) > 1 {
+
+		b := args[1]
+		var err error
+		if !IsJSON(b) {
+			b, err = Jsonify(args[1:])
+			if err != nil {
+				return nil, err
+			}
+			log.Infof("parsed arguments as %v", b)
+		}
+		err = json.Unmarshal([]byte(b), &clone)
+		if err != nil {
+			return nil, fmt.Errorf("invalid payload: %w", err)
+		}
+	}
+	return clone, nil
+}
 
+// hemictlApi is a structure used to satisfy the protocol.API interface.
+type hemictlApi struct {
```

Comment:
Go generally recommend using ALLCAPS for initialisms and acronyms: https://go.dev/wiki/CodeReviewComments#initialisms

```suggestion
type hemictlAPI struct {
```

Reply:
Noted. Fixed 👍